src/Aspire.Hosting/Dcp/ApplicationExecutor.cs

```diff
@@ -1036,23 +1036,29 @@ private void PrepareProjectExecutables()
             else
             {
                 exeSpec.ExecutionType = ExecutionType.Process;
-                if (configuration.GetBool("DOTNET_WATCH") is not true)
+                if (project.TryGetLastAnnotation<DotnetWatchAnnotation>(out var dwa))
                 {
-                    exeSpec.Args = [
-                        "run",
-                        "--no-build",
+                    exeSpec.Args =
+                    [
+                        "watch",
+                        "--non-interactive",
                         "--project",
-                        projectMetadata.ProjectPath,
+                        projectMetadata.ProjectPath
                     ];
+
+                    if (dwa.EnableHotReload)
+                    {
+                        exeSpec.Args.Add("--no-hot-reload");
```

Comment:
This logic seems conflicting. "If enable hot reload" => "--no-hot-reload".

Do we need to rename the property?

Reply:
should be if not enabled -> ...